### PR TITLE
Recorded Future Feed - Add the option to have comma-separated risk rules

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/40609

## Description
Added the option to set a comma-separated list of risk rules as the value of the 'risk_rule' integration's parameter.
If more than one risk rule is setted, the 'fetch-indicators' and 'rf-feed-get-indicators' commands will be executed for each risk rule - i.e the api call is executed for each risk rule separately.  

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShacharKidor** [comment]
Following @reutshal request - I unskipped the recorded future feed test, and the recorded future test in this PR because our quota has been renewed today (01.12.21).

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py [hunk 1]
```diff
@@ -71,20 +71,21 @@ def __init__(self, indicator_type: str, api_token: str, services: list, risk_rul
         self.tlp_color = tlp_color
         super().__init__(self.BASE_URL, proxy=proxy, verify=not insecure)
 
-    def _build_request(self, service, indicator_type):
+    def _build_request(self, service, indicator_type, risk_rule):
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py:74]: please add type hints

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py [hunk 2]
```diff
@@ -112,17 +113,18 @@ def _build_request(self, service, indicator_type):
             raise DemistoException(f'Service unknown: {service}')
         return response.prepare()
 
-    def build_iterator(self, service, indicator_type):
+    def build_iterator(self, service, indicator_type, risk_rule):
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py:116]: please add type hints

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py [hunk 3]
```diff
@@ -246,7 +249,9 @@ def test_module(client: Client, *args) -> Tuple[str, dict, dict]:
     client.run_parameters_validations()
 
     for service in client.services:
-        client.build_iterator(service, client.indicator_type)
+        # if there are risk rules, select the first one for test
+        risk_rule = client.risk_rule[0] if client.risk_rule else None
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py:253]: why dont we test all ?

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py [hunk 4]
```diff
@@ -322,18 +327,35 @@ def format_risk_string(risk_string):
     return f'{splitted_risk_string[0]} of {splitted_risk_string[1]} Risk Rules Triggered'
 
 
-def fetch_indicators_command(client, indicator_type, limit: Optional[int] = None):
-    """Fetches indicators from the Recorded Future feeds.
+def fetch_and_create_indicators(client, risk_rule=None):
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py:330]: please add type hints

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py [hunk 5]
```diff
@@ -322,18 +327,35 @@ def format_risk_string(risk_string):
...
+
+    Returns: None.
+
+    """
+    indicators_batch = fetch_indicators_command(client, client.indicator_type, risk_rule)
+    for indicators in indicators_batch:
+        demisto.createIndicators(indicators)
+
+
+def fetch_indicators_command(client, indicator_type, risk_rule=None, limit: Optional[int] = None):
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py:346]: please add type hints

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.yml [hunk 6]
```diff
@@ -105,9 +105,11 @@ configuration:
...
 - additionalinfo: |-
-    Limit the indicators list to a specific risk rule. To see available
-    risk rules run the rf-feed-get-risk-rules command. This parameter will only be
-    used for the 'connectApi' service.
+    A comma-separated list of risk rules which limit the indicators list to a specific risk rule.
+    For example: 'dhsAis,phishingUrl'.
+    If more than one risk rule is setted, the 'fetch-indicators' and 'rf-feed-get-indicators' commands will be executed for each risk rule.
+    To see available risk rules run the rf-feed-get-risk-rules command.
+    This parameter will only be used for the 'connectApi' service.
     Using the 'large' risk rule is not recommended.
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.yml:113]:
```suggestion
    A comma-separated list of risk rules which limits the indicators list to a specific risk rule.
    For example: 'dhsAis,phishingUrl'.
    If more than one risk rule is set, the indicators fetching and the 'rf-feed-get-indicators' command will be executed for each risk rule.
    To see available risk rules run the rf-feed-get-risk-rules command.
    This parameter will only be used for the 'connectApi' service.
    Using the 'large' risk rule is not recommended.
```

File: Packs/FeedRecordedFuture/ReleaseNotes/1_0_17.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Recorded Future RiskList Feed
+- Added the option to set a comma-separated list of risk rules as the value of the **risk_rule** integration's parameter.
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/ReleaseNotes/1_0_17.md:4]:
```suggestion
- Added the option to set a comma-separated list of risk rules as the value of the **Risk Rule** integration parameter.
```

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import pytest
 from collections import OrderedDict
-from FeedRecordedFuture import get_indicator_type, get_indicators_command, Client, fetch_indicators_command
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py:3]: why did you move the imports into the test funcs?

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py [hunk 9]
```diff
@@ -111,6 +113,51 @@ def test_get_indicators_command(mocker, indicator_type, build_iterator_answer, v
...
+     - Verify the raw response and the human readable output of the command are correct, and include fetched indicators
+      of all the defined risk rules.
+    """
+    from FeedRecordedFuture import get_indicators_command, Client
+    client = Client(indicator_type=indicator_type, api_token='123', risk_rule=risk_rules, services='ConnectApi')
+    args = {
+        'indicator_type': indicator_type,
+        'limit': 1
+    }
+    mocker.patch('FeedRecordedFuture.Client.build_iterator')
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py:148]: can we use `requests_mocks` instead of this ?

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py [hunk 10]
```diff
@@ -190,3 +240,50 @@ def __next__(self):
...
+     2. Error message for invalid risk rule.
+     3. Error message for invalid risk rule on the second risk rule in the risk rules list.
+    """
+    mocker.patch('FeedRecordedFuture.return_error', side_effect=mock_return_error)
+    from FeedRecordedFuture import Client
+    client = Client(indicator_type=indicator_type, api_token='123', risk_rule=risk_rules, services=service)
+
+    mocker.patch('FeedRecordedFuture.Client.get_risk_rules',
+                 return_value={'data': {'results': [{'name': 'dhsAis'}, {'name': 'phishingUrl'}]}})
+    with pytest.raises(Exception) as e:
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py:278]: i think you can catch `SystemExit` and avoid mocking `return_error`

**Review by Itay4 — COMMENTED** [review]
please add to content-test-conf another instance of the feed where there is comma-seperated list of risk rules

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py (hunk 8 above)
Comment (ShacharKidor) [reply Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py:3]: This changed during the writing of the new tests I added. I changed it back.

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py (hunk 3 above)
Comment (ShacharKidor) [reply Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py:253]: the validation of all configured risk rules is verified in run_parameters_validations(). 
Running the fetch flow for each one of them takes time ... I think it is unnecessary and we can be satisfied with running the flow for one of them in order to check the correctness of the instance and the API calls

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py (hunk 9 above)
Comment (ShacharKidor) [reply Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py:148]: > please add to content-test-conf another instance of the feed where there is comma-seperated list of risk rules

@Itay4  - added in this merge request: [1339](https://code.pan.run/xsoar/content-test-conf/-/merge_requests/1339)

**Comment by ShacharKidor** [comment]
> please add to content-test-conf another instance of the feed where there is comma-seperated list of risk rules

@Itay4 - added in this merge request: [1339](https://code.pan.run/xsoar/content-test-conf/-/merge_requests/1339)

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py (hunk 10 above)
Comment (ShacharKidor) [reply Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture_test.py:278]: changed that.

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py [hunk 11]
```diff
@@ -71,20 +71,21 @@ def __init__(self, indicator_type: str, api_token: str, services: list, risk_rul
         self.tlp_color = tlp_color
         super().__init__(self.BASE_URL, proxy=proxy, verify=not insecure)
 
-    def _build_request(self, service, indicator_type):
+    def _build_request(self, service, indicator_type, risk_rule: Optional[str] = None):
```
Comment (Itay4) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py:74]:
```suggestion
    def _build_request(self, service, indicator_type, risk_rule: Optional[str] = None) -> requests.PreparedRequest:
```

**Review by Itay4 — APPROVED** [review]
Alt Text
